examples/javascript-object-detection/index.js

```diff
@@ -0,0 +1,63 @@
+import { pipeline, env } from 'https://cdn.jsdelivr.net/npm/@xenova/transformers@2.5.3';
+env.allowLocalModels = false;
+import { generateRandomColor, removeElements, getScaledCoordinates } from './utils.js'
+const statusParagraph = document.getElementById("status");
+const fileUploadElement = document.getElementById('file-upload');
+const imageContainer = document.getElementById("image-container");
+
+fileUploadElement.addEventListener('change', function(e) {
+    const file = e.target.files[0];
+    const fr = new FileReader();
+    fr.onload = onFileReaderLoad;
+    fr.readAsDataURL(file);
+});
+
+function onFileReaderLoad(e) {
+    removeElements("bounding-box-label");
+    removeElements("bounding-box");
+    const base64String = e.target.result;
+    const imageEl = document.createElement('img');
+    imageEl.src = base64String;
+    imageContainer.appendChild(imageEl);
+    runModel(imageEl);
+};
+
+async function runModel(imageEl) {
+    statusParagraph.textContent = "Loading model...";
+    const detector = await pipeline('object-detection', 'Xenova/detr-resnet-50');
```

Comment:
This creates a new pipeline every time we perform detection (and will keep allocating memory until we run out). I think we can simply move this to the top of the file instead. One downside is that it will start downloading on page load instead of lazily.

Maybe a better option would be to do `let detector;` at the top, and then `if (!detector) detector = await pipeline(...)`.